src/migrations.test.js

```diff
@@ -76,8 +84,9 @@ describe('data migrations', () => {
     const [hasMigrated, res] = await migrateStorageData(migrations, { sites })
 
     expect(hasMigrated).toBe(true)
-    const warningIds = res.warnings.map(warning => warning.id)
-    expect(warningIds).toEqual([0, 1, 2])
+    // Only checking length here because further migrations mess with all the
+    // other fields
+    expect(res.warnings).toHaveLength(3)
```

Comment:
These tests are starting to get really messy. I think that their main issue is that they're trying to test that things from from one migration to the next, but they assume the end migration. As more and more migrations come in, these end up being more and more useless.

I think that we need to test migrations in a way that ensure certain truths regardless of the data version a migration starts at. I think that data driven tests could help here. We could do stuff like:

- Make sure that we start with a sane empty set (from v0, v1, v2, etc.)
- Make sure that fields are converted (from v0, v1, v2, etc.)
- Make sure that we don't get duplicate ids (from v0, v1, v2, etc.)
- Make sure that warnings get some default values if they're not set (from v0, v1, v2, etc.)